src/Components/Common/AzureComponent.cs

```diff
@@ -53,7 +53,15 @@ internal void AddClient(
         var configSection = builder.Configuration.GetSection(configurationSectionName);
 
         var settings = new TSettings();
+        // Bind both top-level and named configuration sections to the settings object
+        // to allow connection-specific settings.
         BindSettingsToConfiguration(settings, configSection);
+        BindSettingsToConfiguration(settings, configSection.GetSection(connectionName));
```

Comment:
Can we add a test to ensure the binding happens correctly? Specifically that the named section overrides the root section.